src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -68,58 +68,52 @@ internal class SqlServerSearchService : SearchService
 
         protected override async Task<SearchResult> SearchInternalAsync(SearchOptions searchOptions, CancellationToken cancellationToken)
         {
-            using (var connection = new SqlConnection(_configuration.ConnectionString))
-            {
-                connection.Open();
+            SearchResult searchResult;
 
-                SearchResult searchResult;
+            // If we should include the total count of matching search results
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+            {
+                searchResult = await SearchImpl(searchOptions, false, cancellationToken);
 
-                // If we should include the total count of matching search results
-                if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+                // If this is the first page and there aren't any more pages
+                if (searchOptions.ContinuationToken == null && searchResult.ContinuationToken == null)
                 {
-                    // Begin a transaction so we can perform two atomic reads.
-                    using (var transaction = connection.BeginTransaction(IsolationLevel.ReadCommitted))
+                    // Count the results on the page.
+                    searchResult.TotalCount = searchResult.Results.Count();
+                }
+                else
+                {
+                    try
                     {
-                        try
-                        {
-                            searchResult = await SearchImpl(searchOptions, false, connection, cancellationToken, transaction);
```

Comment:
I moved this out of the try-block, addressing @johnstairs's [comment](https://github.com/microsoft/fhir-server/pull/731#discussion_r344317975) from my previous PR.